### PR TITLE
[6.11.z] Deprecating api/utils.py module

### LINKED ISSUE #12326: [Failed-AutoCherryPick] - Deprecating api/utils.py module
#### Auto-Cherry-Pick WorkFlow Failure:
- To Branch: 6.11.z
- [Failed Cherrypick Action](https://github.com/SatelliteQE/robottelo/actions/runs/5956816831)
- [Parent Pull Request](https://github.com/SatelliteQE/robottelo/pull/12175)

### PR DESCRIPTION
**Pull request by jyejare**
Closes #12326 

Cherrypick of #12175

### CODE REVIEW

File: tests/foreman/destructive/test_contentview.py [hunk 1]
```diff
@@ -81,14 +80,14 @@ def test_positive_reboot_recover_cv_publish(target_sat, function_entitlement_man
     try:
         publish_task = cv.publish(synchronous=False)
         target_sat.power_control(state='reboot', ensure=True)
-        wait_for_tasks(
+        target_sat.api_factory.wait_for_tasks(
```
Comment (Gauravtalreja1) [inline tests/foreman/destructive/test_contentview.py:83]: Wait for task is part of Capsule mixins, not api_factory, right?
```suggestion
        target_sat.wait_for_tasks(
```

File: tests/foreman/destructive/test_contentview.py [hunk 2]
```diff
@@ -81,14 +80,14 @@ def test_positive_reboot_recover_cv_publish(target_sat, function_entitlement_man
...
             search_query=(f'id = {publish_task["id"]}'),
             search_rate=30,
             max_tries=60,
         )
     except TaskFailedError:
         target_sat.api.ForemanTask().bulk_resume(data={"task_ids": [publish_task['id']]})
-        wait_for_tasks(
+        target_sat.api_factory.wait_for_tasks(
```
Comment (Gauravtalreja1) [inline tests/foreman/destructive/test_contentview.py:90]: Wait for task is part of Capsule mixins, not api_factory, right?
```suggestion
        target_sat.wait_for_tasks(
```

**Review by Gauravtalreja1 — COMMENTED** [review]
Ack, but small change is required

**Comment by jyejare** [comment]
@Gauravtalreja1 Thanks for catching wait_for_tasks parity, now resolved! Please revisit !

**Review by ogajduse — APPROVED** [review]

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**